crowbar_framework/lib/crowbar/sanity.rb

```diff
@@ -0,0 +1,67 @@
+#
+# Copyright 2016, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Crowbar
+  class Sanity
+    class << self
+      def check
+        [].tap do |errors|
```

Comment:
for now we have only the network checks, but it could be more sanity checks in the future. so each check below (which itself calls more sub-checks) can return multiple errors, that is why I chose this design.